src/guide/chapter_3.rs

```diff
@@ -0,0 +1,495 @@
+```
+
+
+Conrod can use graphics. It stores these in a map. The system needs the map,
+even though it doesn't contain anything at this time, so create it:
+
+```txt
+    let image_map = conrod::image::Map::<glium::texture::Texture2d>::new();
+```
+
+Finally, Conrod needs to render its UI. It uses a renderer to do this, so
+create one:
+
+```txt
+    let mut renderer = conrod::backend::glium::Renderer::new(&display).unwrap();
+```
+
+As an Immediate Mode GUI, Conrod sits in the main loop of the program, drawing
+the UI every time round the loop. Here's the main loop:
+
+```txt
+'main: loop {
+    // Render the `Ui` and then display it on the screen.
+    if let Some(primitives) = ui.draw_if_changed() {
+        renderer.fill(&display, primitives, &image_map);
+        let mut target = display.draw();
+        target.clear_color(0.0, 1.0, 0.0, 1.0);
+        renderer.draw(&display, &mut target, &image_map).unwrap();
+        target.finish().unwrap();
+    }
+}
+```
+
+Right now, the program really doesn't do much. This is the minimum set of bits
+and pieces to get a window up on the screen. The UI has nothing in it, so the
+call to `ui.draw_if_changed()` has nothing to do. The UI remembers whether
+there is a need to draw itself, so it does not draw on every cycle of the main
+loop, only if things have changed.
+
+`target = display.draw()` starts drawing on the backbuffer. It returns a
+Frame, which the renderer can draw everything to. The first command to
+`target` is to `clear_color`. This takes fractional red, green and blue values
+between 0.0 and 1.0, and an alpha value, also between 0.0 and 1.0 and fills
+the frame, in this case with bright green. The renderer is then asked to draw
+the rest of the UI (ie nothing) to the frame, and finally, `target`, the
+frame, is told to finish. It stops the drawing, swaps the OpenGL buffers
+(putting the just drawn frame on the screen) and consumes the frame. The frame
+docs are part of glium
+[here](https://docs.rs/glium/0.17.1/glium/struct.Frame.html).
+
+
+If you've run this program, you'll have seen the single least useful output
+ever - a green block with a white "menu bar" that has nothing in it. Let's add
+at least something to see.
+
+After
+# const WIDTH: i32 = 400;
+# const HEIGHT: i32 = 640;
+```txt
+    let mut ui = conrod::UiBuilder::new([WIDTH as f64, HEIGHT as f64]).build();
+```
+
+add the code that will initiate and build the UI:
+
+```txt
+widget_ids!(struct Ids { text });
+let ids = Ids::new(ui.widget_id_generator());
+```
+
+`widget_ids!` is, as its name suggests, a macro provided by conrod to create
+widget ids. Conrod's primary data structure contains all the widgets, and uses
+their ids to keep track of them. The `widget_ids!` macro just provides a very
+quick and easy way of defining widget names (in this case just `text`) and
+giving them ids.
+
+After defining the widget_ids, `Ids::new` creates the the widget structure.
```

Comment:
I was following the guide and noticed a typo here, `creates the the widget structure`